core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/toDataFrame.kt

```diff
@@ -397,4 +398,30 @@ class CreateDataFrameTests {
             DataColumn.createValueColumn("c", listOf(arrayOf(5, null)), typeOf<Array<Int?>>()),
         ).schema()
     }
+
+    @DataSchema
+    data class Person(val firstName: String, val lastName: String, val age: Int, val city: String?) : DataRowSchema
+
+    @DataSchema
+    data class Group(val id: String, val participants: List<Person>) : DataRowSchema
+
+    @Test
+    fun `deeply convert data schema and list of data schema`() {
```

Comment:
Is it worth to compare here, for example, a circular references?

Group depends on Person, Person depends on Group

Reply:
I started working on this PR with this circular reference idea in mind. Because regular Kotlin object can have them quite easily. But since we ended up converting only DataSchema objects deeply, i think it's not an issue anymore. You really need to do it on purpose to have a cycle between DataSchema objects. So i think it's not worth to bother here